MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/javaservices/TurfLineDistanceActivity.java

```diff
@@ -136,16 +137,14 @@ public void onStyleLoaded(@NonNull Style style) {
 
           double distanceBetweenLastAndSecondToLastClickPoint = 0;
 
-          // Make the Turf calculation between the first and second map tap point.
-          // Add that distance to the overall line distance.
-          if (pointList.size() == 2) {
+          if (pointList.size() >= 2) {
+            // If the point list only has two points, make the Turf calculation between
+            // the first and second map tap point. Add that distance to the overall
+            // line distance. Make the Turf calculation between the last tap point and
+            // the second-to-last tap point.
             distanceBetweenLastAndSecondToLastClickPoint = TurfMeasurement.distance(
-              pointList.get(listSize - 1), pointList.get(0));
-          } else if (pointList.size() > 2) {
-            // Make the Turf calculation between the last tap point and the second-to-last
-            // tap point.
-            distanceBetweenLastAndSecondToLastClickPoint = TurfMeasurement.distance(
-              pointList.get(listSize - 2), pointList.get(listSize - 1));
+                pointList.get(listSize - pointList.size() == 2 ? 1 : 2),
+                listSize - pointList.size() == 2 ? pointList.get(0) : pointList.get(listSize - 1));
           }
```

Comment:
I'm thinking this could be simplified to the following:
```
          if (pointList.size() >= 2) {
            // Make the Turf calculation between the last tap point and the second-to-last
            // tap point.
            distanceBetweenLastAndSecondToLastClickPoint = TurfMeasurement.distance(
              pointList.get(listSize - 2), pointList.get(listSize - 1));
          }
```
Since I think maybe the order of two points to be measured doesn't matter inside the `TurfMeasurement.distance(point1, point2)`?

Reply:
🤦‍♂ , yes that works when I open the example and play around with it. Fixed in latest commit.